conan/internal/api/detect_api.py

```diff
@@ -228,6 +230,7 @@ def _mcst_lcc_cppstd_default(version):
     default = {"gcc": _gcc_cppstd_default(compiler_version),
                "clang": _clang_cppstd_default(compiler_version),
                "apple-clang": "gnu98",
+               "intel-cc": "17",
```

Comment:
Any version of intel OneAPI defaults to C++17?
If there is some public official link that defines that, feel free to paste it in code as a comment, that helps for maintenance when new oneapi versions are released.

Reply:
You are perfectly right. Intel oneAPI is switching from C++14 to C++17 [starting from release 2023.0](https://www.intel.com/content/www/us/en/developer/articles/troubleshooting/icx-changes-default-cpp-std-to-cpp17-with-2023.html). Thank you for pointing this out, I will add code to manage this.